runtime/regexp.h

```diff
@@ -263,18 +266,18 @@ string regexp::get_replacement(const T &replace_val, const string &subject, int6
 
 
 template<class T>
-mixed regexp::replace(const T &replace_val, const string &subject, int64_t limit, int64_t &replace_count) const {
+Optional<string> regexp::replace(const T &replace_val, const string &subject, int64_t limit, int64_t &replace_count) const {
   pcre_last_error = 0;
   int64_t result_count = 0;//calls can be recursive, can't write to replace_count directly
 
   check_pattern_compilation_warning();
   if (pcre_regexp == nullptr && RE2_regexp == nullptr) {
-    return mixed();
+    return false;
```

Comment:
Is it ok?
In functions.txt it is written, that this function returns only `string|null` without `false`:
```
function preg_replace (....) ::: ^3|string|null;
```

The same about other `false` below.

Reply:
Ok, now I see below that many of the `f $ preg_replace` functions return `false`. It might make sense to fix the signature in functions.txt.

Reply:
Good observations!
I changed all `return false` to `return {}` to return a proper null there.
Added `002_null_error.php` test to check that this behavior is consistent with PHP.